addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRuleUnitTest.java

```diff
@@ -106,6 +108,23 @@ void shouldNotRaiseAlertGivenIntegrityAttributeIsPresentInLinkElement()
         assertThat(alertsRaised.size(), equalTo(0));
     }
 
+    @Test
+    void shouldNotIndicateElementGivenElementIsServedByTrustedDomains()
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head>"
+                                + "<script src=\"https://some.cdn.com/v1.0/include.js\"></script>"
+                                + "</head><body></body></html>");
+        rule.getConfig()
+                .setProperty(TrustedDomains.TRUSTED_DOMAINS_PROPERTY, "https://some.cdn.com/.*");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
```

Comment:
This causes a NPE for some reason that I haven't been able to sort out.

Reply:
`msg.getHistoryRef() == null` is returning true for some reason (in the scan rule inside `scanHttpResponseReceive`), but only in this particular unit test method.

Reply:
Found it, should have been calling `scanHttpResponseReceive(msg)`